src/PKSim.Core/Commands/EditParameterCommand.cs

```diff
@@ -51,36 +54,40 @@ protected void UpdateDependenciesOnParameter(IExecutionContext context)
       private void udpateBuildingBlockProperties(IExecutionContext context)
       {
          var parentBuildingBlock = context.BuildingBlockContaining(_parameter) ?? context.Get<IPKSimBuildingBlock>(BuildingBlockId);
-         context.UpdateBuildinBlockProperties(this, parentBuildingBlock);
+         context.UpdateBuildinBlockPropertiesInCommand(this, parentBuildingBlock);
       }
 
       protected abstract void UpdateParameter(IParameter parameter, IExecutionContext context);
 
       protected IParameter OriginParameterFor(IParameter parameter, IExecutionContext context)
       {
-         if (!IsBuildingBlockParameterInSimulation(parameter))
+         if (!isBuildingBlockParameterInSimulation(parameter))
             return null;
 
          return context.Get<IParameter>(parameter.Origin.ParameterId);
       }
 
-      protected bool IsBuildingBlockParameterInSimulation(IParameter parameter)
+      private bool isBuildingBlockParameterInSimulation(IParameter parameter)
       {
-         //a parameter defined in a simulation? Not a building block parameter
-         if (parameter.BuildingBlockType == PKSimBuildingBlockType.Simulation) return false;
+         //Building block id is not set => This is the case of parameter defined in a building block
+         if (string.IsNullOrEmpty(parameter.Origin.BuilingBlockId))
+            return false;
```

Comment:
Typo: Buil**d**ingBlockId

Reply:
Well it's a property from the object. I'd rather not have to do project conversion to be honest

Reply:
been like that since forever :)